### ISSUE TITLE
[package] catch2/2.13.6: missing files for CMake find_package()

### ISSUE THREAD

**Issue by scandyna** [conan-io/conan-center-index#6035]
### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: catch2/2.13.6
  * Operating System+version: Linux Ubuntu 18.04
  * Compiler+version: GCC 7
  * Docker image: registry.gitlab.com/scandyna/docker-images/ubuntu-18.04-cpp-gui:latest
  * Conan version: conan 1.37
  * Python version: Python 3.6.9

### Conan profile
```
Configuration for profile linux_gcc7_x86_64:

[settings]
os=Linux
os_build=Linux
arch=x86_64
arch_build=x86_64
compiler=gcc
compiler.version=7
compiler.libcxx=libstdc++11
compiler.cppstd=14
[options]
[build_requires]
[env]
CC=gcc-7
CXX=g++-7
```

### Logs
An example, you could look here: https://gitlab.com/scandyna/mdtapplication/-/jobs/1370197871

### Details

I installed catch2 on my local machine, and package folder.
In `lib/cmake/Catch2/` `Catch2Config.cmake` is missing.

I also have `Catch2/2.11.1@catchorg/stable` installed, and in `lib/cmake/Catch2/` all the files needed for CMake `find_package()` are present. Those files are `Catch2Config.cmake`, `Catch2ConfigVersion.cmake` and `Catch2Targets.cmake`.

Looking in the recipe, I have seen:
```
    def package(self):
        self.copy(pattern="LICENSE.txt", dst="licenses", src=self._source_subfolder)
        cmake = self._configure_cmake()
        cmake.install()
        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
        tools.rmdir(os.path.join(self.package_folder, "share"))
        for cmake_file in ["ParseAndAddCatchTests.cmake", "Catch.cmake", "CatchAddTests.cmake"]:
            self.copy(
                cmake_file,
                src=os.path.join(self._source_subfolder, "contrib"),
                dst=os.path.join("lib", "cmake", "Catch2"),
```

Notice that the required files are generated by CMake at install time. So, the `tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))` is probably a cause of the issue.

**Comment by madebr** [comment]
Most conan recipes of this repos are intended to be used with the `cmake_find_package` or `cmake_find_package_multi` generator. This generator will create `FIndCatch2.cmake` or `Catch2-config.cmake`.

Try adding it to the generators section.

**conanfile.txt**
```txt
[requires]
catch2/2.13.6
[generators]
cmake_find_package
```

**Comment by scandyna** [comment]
I did not use the `cmake` generator until I had to compile with `Clang` and `libc++`. The `cmake` generator solved my problems.
In the `CMakeLists.txt`:
```cmake
if(EXISTS "${CMAKE_BINARY_DIR}/conanbuildinfo.cmake")
  include("${CMAKE_BINARY_DIR}/conanbuildinfo.cmake")
  conan_basic_setup(NO_OUTPUT_DIRS)
endif()
```

Also, the `cmake` generator adds all the required dependencies to `CMAKE_PREFIX_PATH`, which makes `find_package()` work out of the box.

I did not try `cmake_find_package` now.
Does it generate [find-modules](https://cmake.org/cmake/help/v3.21/manual/cmake-developer.7.html#find-modules) ?

I'm not a `Conan` expert at all, neither a `CMake` expert, but have enough experience to know that `CMake` based libraries should provide their own [Config-file Packages](https://cmake.org/cmake/help/v3.21/manual/cmake-packages.7.html), which `Catch2` does. Some package files are generated by `CMake`, and provides some `Import target` with required properties attached.
Do you aim to reproduce the (sometimes very) complex logic of `CMake` install ?

**Comment by madebr** [comment]
Yes, `cmake_find_package` creates `FindXXX.cmake` files `cmake_find_package_multi` created `XXXConfig.cmake` files.
Currently, they are the (only) way to get `find_package` working with the packages on cci.
(But know that going to conan 2.0, these generators will be deprecated)

What cci generally does is remove the often non-relocatable files and autogenerate these.
Advanced functionality (such as the catch2 functions) are made available by copying the files containing them.
See https://github.com/conan-io/conan-center-index/blob/160f54bef9b7931000a37abe6750e4372d6dbd53/recipes/catch2/2.x.x/conanfile.py#L74-L79
+
https://github.com/conan-io/conan-center-index/blob/160f54bef9b7931000a37abe6750e4372d6dbd53/recipes/catch2/2.x.x/conanfile.py#L93
where this is done.
